torch/_dynamo/variables/functions.py

```diff
@@ -831,12 +831,36 @@ def __init__(self, kernel, kernel_idx, grid, **kwargs):
     def call_function(
         self, tx, args: "List[VariableTracker]", kwargs: "Dict[str, VariableTracker]"
     ) -> "VariableTracker":
-        from triton.runtime.autotuner import Autotuner
+        from triton.runtime.autotuner import autotune, Autotuner, Config
+
+        from torch._higher_order_ops.triton_kernel_wrap import kernel_side_table
 
         from .constant import ConstantVariable
         from .dicts import ConstDictVariable
         from .lists import BaseListVariable
 
+        special_kwargs = {}
+        for name in ("num_warps", "num_stages", "num_ctas"):
```

Comment:
Looks like we dont actually read num_ctas from triton.config
https://github.com/pytorch/pytorch/blob/e7141d117f5f870dab61e3a991bf05dab61ed220/torch/_inductor/codegen/wrapper.py#L1117-L1124

Reply:
You should also check whether inductor autotuner actually reads the value of num_ctas and does the right thing with it

Reply:
Inductor doesn't seem to render `num_ctas` currently in the codegen. Still, I think, we should move `num_ctas` to `Config` in this PR, and possibly add rendering `num_ctas` in the Inductor codegen as a follow-up.

Reply:
For now, raising `Unsupported` if `num_ctas` is passed as a direct kernel argument.